front/gatsby/src/pages/index.tsx

```diff
@@ -1,29 +1,40 @@
-// Copyright 2018-2019 @paritytech/substrate-light-ui authors & contributors
+// Copyright 2018-2019 @paritytech/nomidot authors & contributors
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { ApiContext, ApiContextType } from '@substrate/context/src';
-import React from 'react';
+import { Loading } from '@substrate/ui-components/src';
+import React, { useEffect, useState } from 'react';
 
+// global
 import Layout from '../components/layout';
 import SEO from '../components/seo';
+// context
 import { ContextGate } from '../ContextGate';
-import { Onboarding } from './onboarding';
+import Onboarding from './onboarding';
+// pages
+import Profile from './profile';
 
 function IndexPage(): React.ReactElement {
-  // const { injectedAccounts } = useContext(AccountsContext);
+  const [exploringAs, setExploringAs] = useState();
+
+  useEffect(() => {
+    setExploringAs(localStorage.getItem('exploringAs'));
+  }, []);
 
   return (
     <ContextGate>
       <ApiContext.Consumer>
         {({
           isReady,
         }: Partial<ApiContextType>): React.ReactElement | boolean | undefined =>
-          isReady && (
+          isReady ? (
             <Layout>
               <SEO title='Home' />
-              <Onboarding />
+              {exploringAs ? <Profile /> : <Onboarding />}
             </Layout>
+          ) : (
+            <Loading active inline />
```

Comment:
`<Loading active inline />` should go inside `ApiContextProvider` in the context gate, as the `loading` prop. And then you won't need the consumer here anymore.